src/utils/vcs.rs

```diff
@@ -249,6 +249,54 @@ pub fn git_repo_head_ref(repo: &git2::Repository) -> Result<String> {
     }
 }
 
+pub fn git_repo_base_ref(repo: &git2::Repository, remote_name: &str) -> Result<Option<String>> {
+    // Get the current HEAD commit
+    let head_commit = repo.head()?.peel_to_commit()?;
+
+    // Try to find the remote tracking branch
+    let remote_branch_name = format!("refs/remotes/{remote_name}/HEAD");
+    let remote_ref = match repo.find_reference(&remote_branch_name) {
+        Ok(r) => r,
+        Err(_) => {
+            // If remote/HEAD doesn't exist, try to find the default branch
+            // First try common default branch names
+            for branch in &["main", "master", "develop"] {
```

Comment:
Is there a direct way to detect the default branch, besides just checking against hardcoded common values for them? This implementation seems flimsy and easy to break, e.g. if a repo has all three of these branch names in it.

If it is the best we can do, I guess it is okay, but I'd rather directly check for the default branch than just guess it